dynamic_programming/coin_change.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Given a set of categories of coins C and an amount of money S, the goal is:
+ * to give change for S but to use a minimum number of coins. Suppose each category of coin has an infinite number of pieces.
+ * @param money - amon of money.
+ * @param coins - The coins that are available.
+ * @returns CoinChange, the minimum number of coins, and which coins are selected
+ */
+
+export interface CoinChange {
+    minCoins: number,
+    coins: number[]
+}
+
+export const coinChange = (money: number, coins: number[]): CoinChange => {
+
+    const dp: number[] = Array(money + 1).fill(1000000000);
```

Comment:
Please give this variable a descriptive name, such as `minCoins`.

Reply:
done